File: lib/platform_specifics/android_styles/inbox_style_information.dart
import 'package:flutter_local_notifications/platform_specifics/android_styles/default_style_information.dart';

class InboxStyleInformation extends DefaultStyleInformation {
  /// Overrides ContentTitle in the big form of the template.
  final String contentTitle;

  /// Set the first line of text after the detail section in the big form of the template.
  final String summaryText;

  /// The lines that form part of the digest section for inbox-style notifications
  List<String> lines;

  /// Specifies if the lines should have formatting applied through HTML markup
  final bool htmlFormatLines;

  /// Specifies if the overridden ContentTitle should have formatting applied through HTML markup
  final bool htmlFormatContentTitle;

  /// Specifies if formatting should be applied to the first line of text after the detail section in the big form of the template.
  final bool htmlFormatSummaryText;

  InboxStyleInformation(this.lines,
      {this.htmlFormatLines = false,
      this.contentTitle,
      this.htmlFormatContentTitle = false,
      this.summaryText,
      this.htmlFormatSummaryText = false,
      bool htmlFormatContent = false,
      bool htmlFormatTitle = false})
      : super(htmlFormatContent, htmlFormatTitle);

  Map<String, dynamic> toJson() {
    var styleJson = super.toJson();

    var bigTextStyleJson = <String, dynamic>{
      'contentTitle': contentTitle,
      'htmlFormatContentTitle': htmlFormatContentTitle,
      'summaryText': summaryText,
      'htmlFormatSummaryText': htmlFormatSummaryText,
      'lines': lines ?? new List<String>(),
      'htmlFormatLines': htmlFormatLines
    };
    styleJson.addAll(bigTextStyleJson);
    return styleJson;
  }
}

File: flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/models/styles/InboxStyleInformation.java
package com.dexterous.flutterlocalnotifications.models.styles;

import androidx.annotation.Keep;

import java.util.ArrayList;

@Keep
public class InboxStyleInformation extends DefaultStyleInformation {
  public Boolean htmlFormatLines;
  public ArrayList<String> lines;
  public String contentTitle;
  public Boolean htmlFormatContentTitle;
  public String summaryText;
  public Boolean htmlFormatSummaryText;

  public InboxStyleInformation(
      Boolean htmlFormatTitle,
      Boolean htmlFormatBody,
      String contentTitle,
      Boolean htmlFormatContentTitle,
      String summaryText,
      Boolean htmlFormatSummaryText,
      ArrayList<String> lines,
      Boolean htmlFormatLines) {
    super(htmlFormatTitle, htmlFormatBody);
    this.contentTitle = contentTitle;
    this.htmlFormatContentTitle = htmlFormatContentTitle;
    this.summaryText = summaryText;
    this.htmlFormatSummaryText = htmlFormatSummaryText;
    this.lines = lines;
    this.htmlFormatLines = htmlFormatLines;
  }
}
